packages/universal-router-sdk/src/entities/protocols/pancakeswap.ts

```diff
@@ -0,0 +1,436 @@
+import { TradeType, validateAndParseAddress } from '@pancakeswap/sdk'
+import {
+  BaseRoute,
+  PoolType,
+  RouteType,
+  SmartRouter,
+  SmartRouterTrade,
+  StablePool,
+  SwapRouter,
+  getPoolAddress,
+} from '@pancakeswap/smart-router/evm'
+import { Address } from 'viem'
+import { ROUTER_AS_RECIPIENT, SENDER_AS_RECIPIENT } from '../../utils/constants'
+import { encodeFeeBips } from '../../utils/numbers'
+import { CommandType, RoutePlanner } from '../../utils/routerCommands'
+import { ABIParametersType, AnyTradeType, PancakeSwapOptions } from '../../utils/types'
+import { Command, RouterTradeType, TradeConfig } from '../Command'
+
+// Wrapper for pancakeswap router-sdk trade entity to encode swaps for Universal Router
+// also translates trade objects from previous (v2, v3) SDKs
+export class PancakeSwapTrade implements Command {
+  readonly tradeType: RouterTradeType = RouterTradeType.PancakeSwapTrade
+
+  readonly type: TradeType
+
+  constructor(public trade: AnyTradeType, public options: PancakeSwapOptions) {
+    this.type = (this.trade as SmartRouterTrade<TradeType>).tradeType
```

Comment:
the trade could be an array?

Reply:
try not to use type assertion if possible

Reply:
yep trade could be an array we handle that case in logic below. we type assertion here because smartRouterTradeType cant be assigned to any trade type